src/pocketmine/network/mcpe/protocol/StructureBlockUpdatePacket.php

```diff
@@ -26,16 +26,32 @@
 #include <rules/DataPacket.h>
 
 use pocketmine\network\mcpe\NetworkSession;
+use pocketmine\network\mcpe\protocol\types\StructureEditorData;
 
-class StructureBlockUpdatePacket extends DataPacket{
+class StructureBlockUpdatePacket extends DataPacket/* implements ServerboundPacket*/{
 	public const NETWORK_ID = ProtocolInfo::STRUCTURE_BLOCK_UPDATE_PACKET;
 
+	/** @var int */
+	public $x;
+	/** @var int */
+	public $y;
+	/** @var int */
+	public $z;
+	/** @var StructureEditorData */
+	public $structureEditorData;
+	/** @var bool */
+	public $isPowered;
+
 	protected function decodePayload(){
-		//TODO
+		$this->getBlockPosition($this->x, $this->y, $this->z);
+		$this->structureEditorData = $this->getStructureEditorData();
+		$this->isPowered = $this->getBool();
```

Comment:
There is another bool here. i haven't been able to figure out what it is used for though

Reply:
You're wrong.

Yes, he is, but starting with protocol 370.

Reply:
The bool is still missing here. If not included, you have unread bytes errors

Reply:
The bool is still missing here. If not included, you have unread bytes errors